cmd/server.go

```diff
@@ -172,19 +172,12 @@ func NewServerCommand(stopCh <-chan struct{}) *cobra.Command {
 				// the delta snapshot memory limit), after which a full snapshot
 				// is taken and the regular snapshot schedule comes into effect.
 
-				var startWithFullSnapshot bool
+				var initialDeltaSnapshotTaken bool
 
 				// TODO: write code to find out if prev full snapshot is older than it is
 				// supposed to be, according to the given cron schedule, instead of the
 				// hard-coded "24 hours" full snapshot interval
-				if ssr.PrevFullSnapshot == nil || time.Now().Sub(ssr.PrevFullSnapshot.CreatedOn).Hours() > 24 {
-					startWithFullSnapshot = false
-					if err = ssr.TakeFullSnapshotAndResetTimer(); err != nil {
-						logger.Errorf("Failed to take first full snapshot: %v", err)
-						continue
-					}
-				} else {
-					startWithFullSnapshot = true
+				if ssr.PrevFullSnapshot != nil && time.Now().Sub(ssr.PrevFullSnapshot.CreatedOn).Hours() <= 24 {
```

Comment:
Please change this to `time.Since(ssr.PrevFullSnapshot.CreatedOn)`.

Reply:
Done